src/main/python/integratedPython.ts

```diff
@@ -61,6 +76,21 @@ export const getIntegratedPython = async (
     onProgress: (percentage: number, stage: 'download' | 'extract') => void
 ): Promise<PythonInfo> => {
     const platform = getPlatform();
+    const legacyPythonPath = path.resolve(
+        path.join(directory, get39ExecutableRelativePath(platform))
+    );
+
+    const legacyPythonBinExists = await checkFileExists(legacyPythonPath);
+
+    if (legacyPythonBinExists) {
+        const confirmLegacyPythonVersion = await getPythonVersion(legacyPythonPath);
+        if (gt(confirmLegacyPythonVersion, '3.9.0') && lt(confirmLegacyPythonVersion, '3.10.0')) {
+            const legacyPythonFolder = path.resolve(path.join(directory, '/python'));
+            // Remove legacy integrated python
+            await fs.rm(legacyPythonFolder, { recursive: true, force: true });
+        }
+    }
```

Comment:
This check is kinda dumb.

1. On linux and mac, `checkFileExists` is enough to verify that it is an old version.
2. On windows, we just throw away the python version, and do all these checks (file exists, get version) again.

---

Frankly, I don't think that this approach is all that good in general. I doubt that this will be the only python migration we'll ever do, so we might as well set up a system that makes it easy for us to migrate. How about this:

1. Since the download links are hard-coded, hard-code the python version as well.
2. Optimistically check whether a python exe exists and whether its version is the hard-coded version.
    1. If yes, return the python info.
    2. If no, go to step 3.
3. Delete `<app dir>/python/python` and download the integrated python.
4. Check for the python exe again.

The main advantages of this approach are that 1) we can completely ignore the file structure of old python installations and 2) we do the minimal amount of work if the correct python version is installed.